src/main/java/org/candlepin/subscriptions/product/OfferingSyncController.java

```diff
@@ -90,9 +106,34 @@ public void syncOffering(Offering newState) {
     // taskManager.reconcileCapacityForOffering(...); //NOSONAR
   }
 
+  /**
+   * Enqueues all offerings listed in the product allowlist to be synced with upstream.
+   *
+   * @return number of enqueued products
+   */
+  public int syncAllOfferings() {
```

Comment:
Might be a good idea to get the timing for this process into prometheus so that we can monitor it over time. This can be done by annotating the method and giving it some descriptive metric name. Something like this:

```
@Timed("rhsm-subscriptions.offering.sync")
```

Reply:
On one hand, using the `@Timed` annotation is nice because it is the least amount of coding. I used the Timer micrometer class based on a [recommendation from a previous PR](https://github.com/RedHatInsights/rhsm-subscriptions/pull/551#discussion_r694216471) (on periodically syncing subscriptions) because it would still do the logging to prometheus, AND also because I could log an event that we could then splunk.

Since the already-in-use Timer will get the timing into prometheus, is it alright to keep this as is?

Reply:
> On one hand, using the `@Timed` annotation is nice because it is the least amount of coding. I used the Timer micrometer class based on a [recommendation from a previous PR](https://github.com/RedHatInsights/rhsm-subscriptions/pull/551#discussion_r694216471) (on periodically syncing subscriptions) because it would still do the logging to prometheus, AND also because I could log an event that we could then splunk.
> 
> Since the already-in-use Timer will get the timing into prometheus, is it alright to keep this as is?

This is fine as is. Somehow I totally missed that you were using a timer here. My bad.